client/app/src/Controller/Report/MoneyInController.php

```diff
@@ -64,6 +64,76 @@ public function startAction($reportId)
         ];
     }
 
+    /**
+     * @Route("/report/{reportId}/money-in/exist", name="does_money_in_exist")
+     * @Template("@App/Report/MoneyIn/exist.html.twig")
+     *
+     * @return array|RedirectResponse
+     */
+    public function existAction(Request $request, $reportId)
+    {
+        $report = $this->reportApi->getReportIfNotSubmitted($reportId, self::$jmsGroups);
+        $form = $this->createForm(FormDir\Report\DoesMoneyInExistType::class, $report);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $report = $form->getData();
+            $answer = $form['doesMoneyInExist']->getData();
+
+            $report->setMoneyInExists($answer);
+            $this->restClient->put('report/'.$reportId, $report, ['doesMoneyInExist']);
+
+            if ('yes' === $answer) {
+                return $this->redirectToRoute('money_in_step', ['reportId' => $reportId, 'step' => 1, 'from' => 'does_money_in_exist']);
+            } else {
+                return $this->redirectToRoute('no_money_in_exists', ['reportId' => $reportId, 'from' => 'does_money_in_exist']);
+            }
+        }
+
+        $backLink = $this->generateUrl('money_in', ['reportId' => $reportId]);
+
+        return [
+            'backLink' => $backLink,
+            'report' => $report,
+            'form' => $form->createView(),
+        ];
+    }
+
+    /**
+     * @Route("/report/{reportId}/money-in/no-money-in-exists", name="no_money_in_exists")
+     * @Template("@App/Report/MoneyIn/noMoneyInToReport.html.twig")
+     *
+     * @return array|RedirectResponse
+     */
+    public function noMoneyInToReport(Request $request, $reportId)
+    {
+        $report = $this->reportApi->getReportIfNotSubmitted($reportId, self::$jmsGroups);
+        $form = $this->createForm(
+            FormDir\Report\NoMoneyInType::class,
+            $report,
+            ['translation_domain' => 'report-money-in']);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $report = $form->getData();
+            $answer = $form['reasonForNoMoneyIn']->getData();
+
+            $report->setReasonForNoMoneyIn($answer);
+            $report->getStatus()->setMoneyInState(Status::STATE_DONE);
```

Comment:
I started looking into how we set and retrieve the states for the different sections, but wasn't able to find any place where we actually set the state of a section to incomplete or done.

Might need to figure out how this is done for other sections to make sure we are being consistent. 

This isn't bad/wrong as far as I can tell though.

Reply:
Hey @Gugandeep , thanks for your feedback.

I haven’t actioned this as part of this ticket as I believe that me and Nil will action this as part of the wiring up ticket as my changes will tie in with the updated summary page which will complete the user flow and allow us to update the status update for that section.

I think I had to do a similar thing for the significant decisions work, so I’ll probably revisit that PR to check how the sections state is updated